extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/config/ConfigProperties.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.arc.runtime.config;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Allow configuration properties with a common prefix to be grouped into a single class
+ */
+@Target({ TYPE })
+@Retention(RUNTIME)
+public @interface ConfigProperties {
+
+    String prefix();
```

Comment:
I think that it should be possible to extract the prefix from the class name for simple use cases.

Reply:
Certainly. 

I am thinking of the following simple algorithm: 

* Remove any `Properties`, `Configuration`, `Config` suffix
* Make the remainder lowercase and use tha result.

WDYT?

Reply:
"Make the remainder lowercase" should be something like "Make the remainder lowercase and insert `-` before the positions that previously contained a first uppercase character" or something like that. I think this already happens somewhere in Quarkus, so code for this should already exist.

Reply:
Thanks for pointing it out. I'll check

Reply:
@IIUC you are proposing that `GreetingSubConfiguration` would translate into `greeting.sub` being used as the prefix, correct?

Reply:
To `greeting-sub`; I think that's what Quarkus does elsewhere? But `greeting.sub` would also work.

Reply:
Quarkus does the former, so let's go with it :)

Reply:
Here is the link: https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/configuration/ConfigDefinition.java#L199

Reply:
Yea, that's it! :-)

Reply:
Fixed